Assets/Scripts/Behaviours/AvoidanceBehaviour.cs

```diff
@@ -0,0 +1,32 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+[CreateAssetMenu(menuName = "Flock/Behaviour/Avoidance")]
+public class AvoidanceBehaviour : FilteredFlockBehaviour
+{
+    public override Vector3 CalculateMove(FlockAgent agent, List<Transform> context, Flock flock)
+    {
+        List<Transform> filteredContext = (filter == null) ? context : filter.Filter(agent, context);
+        if (filteredContext.Count == 0)
+            return Vector3.zero;
```

Comment:
I'm guessing this code will be redundant across all implementations of `CalculateMove` right? Maybe you could move this into a shared method in the parent class (`FilteredFlockBehaviour`) which would call the overridden method containing the implementation of the children classes?

Reply:
`FilteredFlockBehaviour` is an abstract class

Reply:
yup but afaik you can still implement methods in an abstract class

Reply:
you could even implement CalculateMove directly in the abstract class (which wouldn't need to be abstract anymore I guess) and just have the initialization stuff in there and then override the method in children classes (and call the base method inside them).

Reply:
I thought about it but even if it's implemented in an overridden `CalculateMove(...)` in `FilteredFlockBehaviour`, you're still gonna have `base.CalculateMove(...)` repeated in all implementations. It's not really any more readable and calling the parent function can add to the overhead.. I think it's fine we leave it like this

Reply:
yeah I think what I would do is to implement `CalculateMove` in `FilteredFlockBehaviour` with the common code, then add a `CalculateFilteredMove` abstract method in `FilteredFlockBehaviour` as well which gets implemented in children classes and gets called inside `CalculateMove`. That way you won't have any redundancy in the code anymore.

Reply:
Thanks for the suggestion, just implemented it and tested.